sources/Outputs/src/ParVRRemote.cpp

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2023, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+#include "ParVRRemote.h"
+
+#include "ParCommon.h"
+
+namespace dfl {
+namespace outputs {
+
+void
+ParVRRemote::writeVRRemotes(boost::shared_ptr<parameters::ParametersSetCollection>& paramSetCollection) {
+  std::unordered_map<algo::GeneratorDefinitionAlgorithm::BusId, bool> busesRegulatedBySeveralGeneratorsToFrozen;
+  for (const auto &keyValue : busesRegulatedBySeveralGenerators_) {
+    busesRegulatedBySeveralGeneratorsToFrozen.insert({keyValue.first, true});
+  }
+
+  for (const auto &generator : generatorDefinitions_) {
+    // if network model, nothing to do
+    if (generator.isNetwork()) {
+      continue;
+    }
+    if (busesRegulatedBySeveralGeneratorsToFrozen.find(generator.regulatedBusId) != busesRegulatedBySeveralGeneratorsToFrozen.end() &&
+        generator.q < generator.qmax && generator.q > generator.qmin) {
+      busesRegulatedBySeveralGeneratorsToFrozen[generator.regulatedBusId] = false;
+    }
+  }
+
+  // delete BusId already in busesRegulatedBySeveralGenerators_ to not generate VRRemotes twice
+  algo::HVDCLineDefinitions::BusVSCMap vscBusVSCDefinitionsMapWithoutBusDuplicates(vscBusVSCDefinitionsMap_);
+  for (const auto& keyValue : busesRegulatedBySeveralGenerators_) {
+    if (vscBusVSCDefinitionsMapWithoutBusDuplicates.find(keyValue.first) != vscBusVSCDefinitionsMapWithoutBusDuplicates.end()) {
+      algo::HVDCDefinition::BusId busId = static_cast<algo::HVDCDefinition::BusId>(keyValue.first);
+      vscBusVSCDefinitionsMapWithoutBusDuplicates.erase(busId);
```

Comment:
Why do we need an intermediate container? We could do the test on the fly below?

Reply:
We can but I thought it could be risky to alter directly vscBusVSCDefinitionsMap_ ^^'

Reply:
This is not my proposal. I am proposing to test if the bus was already implemented directly in the hvdc loop

Reply:
Yes, you are right. I could be optimized.